13.0/s2i/bin/assemble

```diff
@@ -171,7 +172,43 @@ function configure_proxy() {
   xml="$xml\
        </proxy>"
     sed -i "s|<!-- ### configured http proxy ### -->|$xml|" $HOME/.m2/settings.xml
+    return
   fi
+
+  # if http(s)_proxy is set, use that.
+  proxy=${HTTPS_PROXY:-${https_proxy:=${HTTP_PROXY:=$http_proxy}}}
+  noproxy=${NO_PROXY:-$no_proxy}
+  if [ -n $proxy ]; then
+    proxyhost=$(echo $proxy | cut -d : -f 1,2)
+    proxyport=$(echo $proxy | cut -d : -f 3)
+    proxyprotocol="http"
```

Comment:
Could either move the proxyport=${proxyport:-80} here as the default like you did http, or move this proxyprotocol="http" inside the else below?  Would be easier to read the defaults in one place.

Reply:
> Could either move the proxyport=${proxyport:-80} here as the default 

no because then 
`proxyport=${proxyport:-443}`

wouldn't work anymore (it would leave the proxyport set to 80)

> move this proxyprotocol="http" inside the else below

if it makes you happy.

Reply:
(I didn't do it that way initially because it offends my variable scoping sensibilities, but bash doesn't seem to care)